platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineRegionMetadata.java

```diff
@@ -0,0 +1,65 @@
+package com.mapbox.mapboxsdk.offline;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+/**
+ * Arbitrary binary region metadata. The contents are opaque to the mbgl implementation;
+ * it just stores and retrieves a BLOB. SDK bindings should leave the interpretation of
+ * this data up to the application; they _should not_ enforce a higher-level data format.
+ * In the future we want offline database to be portable across target platforms, and a
+ * platform-specific metadata format would prevent that.
+ */
+public class OfflineRegionMetadata {
```

Comment:
My thinking to keep this independent is twofold: 1) let devs use the class as a base class for their higher level implementation, and 2) provide some utils to help with the `byte[]` conversion (see `serialize` and `deserialize`).